### PR TITLE
Add comment in file.cr

### PR DESCRIPTION
**Pull request by MatthieuBlm**
Hi ! I've added comment for method expand_path, open and rename.
Hope it's helpful !

### CODE REVIEW

File: src/file.cr [hunk 1]
```diff
@@ -355,6 +364,11 @@ class File < IO::FileDescriptor
     (stat.st_mode & LibC::S_IFMT) == LibC::S_IFLNK
   end
 
+   # Opens the file named by filename. If a file is being created, its initial
+   # permissions may be set using the perm parameter.
+   # If the optional code block is given, it will be passed the opened file as
```
Comment (asterite) [inline src/file.cr]: No need to talk about the optional block. Methods with blocks are different overloads, so this should be documented in the method that yields.

**Comment by asterite** [comment]
@MatthieuBlm Thank you! This is very helpful indeed, docs help **a lot**. I just made a small comment, after changing that I'll merge.

File: src/file.cr (hunk 1 above)
Comment (MatthieuBlm) [inline src/file.cr]: I've made those changes, tell me if it's what you expected :)

Comment (asterite) [inline src/file.cr]: Almost there :-)

Here the block is not optional, it's the overload that accepts a block. So the docs should probably say something like "opens the file..., yields it to the block, ..., closes the file at the end of the block"

**Comment by asterite** [comment]
@MatthieuBlm Thanks! Could you squash into a single commit?

**Comment by MatthieuBlm** [comment]
@asterite Done, I've squash into the first commit :-)

**Pull request merged**